agenta-web/src/lib/helpers/utils.ts

```diff
@@ -26,25 +44,45 @@ export const EvaluationTypeLabels: Record<EvaluationType, string> = {
     [EvaluationType.auto_webhook_test]: "Webhook Test",
 }
 
-export const saveOpenAIKey = (key: string) => {
+export const saveLlmProviderKey = (providerIdx: number, keyValue: string) => {
     if (typeof window !== "undefined") {
         // TODO: add encryption here
-        localStorage.setItem(openAItoken, key)
+        const keys = JSON.parse(localStorage.getItem(llmAvailableProvidersToken) ?? "[{}]")
+        keys[providerIdx].key = keyValue
+        localStorage.setItem(llmAvailableProvidersToken, JSON.stringify(keys))
     }
 }
 
-export const getOpenAIKey = (): string => {
+export const getLlmProviderKey = (providerName: string) => {
```

Comment:
What I meant was to implement this function like this:

```javascript
export const getLlmProviderKey = (providerKey: string) => getAllProviderLlmKeys().find((item: LlmProvider) => item.key === providerKey)
```

After that, you'll also have to update usage of `getLlmProviderKey` every where by passing in the provider's key in the argument instead of the title

Reply:
Oh I see, thanks for the code snippet, I'll implement it and make the necessary changes

Reply:
I made the changes, but I don't think that finding an provider by its key is the way to go, since what we want for this function is the key itself, or am I missing the point? What I mean is, for example:

In the file agenta-web/src/components/EvaluationTable/AICritiqueEvaluationTable.tsx, in the Line 275 where he goes ``getLlmProviderKey("OpenAI")`` he gets the key by passing the title of the provider so it can get the key, so it doesn't know the key yet.
But again, I might be missing the point, please clarify it for me, and thanks for your time!

Reply:
@MohammedMaaz

Reply:
@aakrem based on the discution so far, and the code, could you provide the answer?